examples/Example17/src/PrimaryGeneratorMessenger.cc

```diff
@@ -55,8 +56,13 @@ PrimaryGeneratorMessenger::PrimaryGeneratorMessenger(PrimaryGeneratorAction *Gun
   fRandomizeGunCmd->AvailableForStates(G4State_Idle);
 
   fAddParticleCmd = new G4UIcmdWithAString("/example17/gun/addParticle", this);
-  fAddParticleCmd->SetGuidance("When using randomization, add a particle to the list of possibilities, with the option "
-                               "to provide a weight to indicate how often it should appear");
+  fAddParticleCmd->SetGuidance("When using randomization, add a particle to the list of possibilities\n\
+                                Usage: /example17/gun/addParticle [type] [weight] [energy] [energy unit]\n\
+                                type: particle name\n\
+                                weight: probability that the particle will appear, between 0 and 1\n\
+                                energy: energy for this type of particle\n\
+                                Note: The default unit applies to the energy parameter.\n\
+                                Note: In order to specify an energy but no weight, use a weight of -1\n");
```

Comment:
OK, I see what -1 means. To make it nicer, during the argument tokenization you could enforce pairs key+value, so if a key like `weight` is missing you would not need to put -1.

Reply:
I can change it to key-value yes, it should be more intuitive than now